### PR TITLE
Use Profile Guided Optimization(pgo) in dotnet core to PreJIT methods…

### PR DESCRIPTION
**Pull request by safihamid**
… during placeholder mode.

Addresses https://github.com/Azure/azure-functions-host/issues/5336

### CODE REVIEW

File: src/WebJobs.Script.WebHost/PreJIT/JitTraceRuntime.cs [hunk 1]
```diff
@@ -0,0 +1,273 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (safihamid) [inline src/WebJobs.Script.WebHost/PreJIT/JitTraceRuntime.cs]: This file is 'As is' from dotnet core team.

**Comment by safihamid** [comment]
@davidwrighton

**Comment by safihamid** [comment]
I am not the best place to document how to generate the .jittrace file. Should I put it somewhere on OneNote?

**Comment by brettsam** [comment]
Does it use public tools? If so, I'd say to add a readme.md file right in the repo next to it... If private tools, then yeah OneNote probably.

File: src/WebJobs.Script.WebHost/PreJIT/coldstart.jittrace [hunk 2]
```diff
@@ -0,0 +1,2086 @@
+4714.7323-[System.Diagnostics.DiagnosticSource]System.Diagnostics.DiagnosticSourceEventSource+TransformSpec+PropertySpec+PropertyFetch+RefTypedFetchProperty`2<System.__Canon,System.__Canon>..ctor(Type,PropertyInfo)~System.Diagnostics.DiagnosticSourceEventSource+TransformSpec+PropertySpec+PropertyFetch+RefTypedFetchProperty`2[[System.__Canon,System.Private.CoreLib],[System.__Canon,System.Private.CoreLib]],System.Diagnostics.DiagnosticSource~2~0~.ctor
```
Comment (davidwrighton) [inline src/WebJobs.Script.WebHost/PreJIT/coldstart.jittrace]: I'd like to see the timestamps removed from the beginning of each line, and then the contents sorted. As written here, each time you update, you'll effectively replace this entire file, which is both big, and hard to analyze.

**Comment by safihamid** [comment]
@brettsam the tools are private until at least dotnet core 5.0. I will add the details in OneNote and send an email with details.

File: src/WebJobs.Script.WebHost/PreJIT/coldstart.jittrace (hunk 2 above)
Comment (safihamid) [reply src/WebJobs.Script.WebHost/PreJIT/coldstart.jittrace]: Good point. it's updated now.

**Comment by fabiocav** [comment]
@safihamid would you flag us here when this is ready for review?

**Comment by safihamid** [comment]
@fabiocav it is ready

File: src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs [hunk 3]
```diff
@@ -15,25 +22,46 @@ public class HostWarmupMiddleware
...
+
+            var file = new FileInfo(path);
+
+            if (file.Exists)
+            {
+                JitTraceRuntime.Prepare(file, out int successfulPrepares, out int failedPrepares);
+
+                // We will need to monitor failed vs success prepares and if the failures increase, it means code paths have diverged or there have been updates on dotnet core side.
+                // When this happens, we will need to regenerate the coldstart.jittrace file.
+                _logger.LogInformation($"PreJIT Successful prepares: {successfulPrepares}, Failed prepares: {failedPrepares}");
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Middleware/HostWarmupMiddleware.cs]: Just one nit on this -- can you add an `EventId` -- something like `new EventId(100, "PreJITResults")`? That will allow us to more easily search for the EventName "PreJITResults" in Kusto.

` _logger.LogInformation(new EventId(100, "PreJIT"), $"PreJIT Successful prepares: {successfulPrepares}, Failed prepares: {failedPrepares}");`

**Pull request merged**